cmd/net.go

```diff
@@ -0,0 +1,106 @@
+/*
+ * Minio Cloud Storage, (C) 2017 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+
+	"github.com/minio/minio-go/pkg/set"
+)
+
+const defaultPort = "9000"
+
+// splitHostPort is wrapper to net.SplitHostPort(), but ignores "missing port in address".
+// If port is missing, default port "9000" is returned.
+func splitHostPort(hostPort string) (host, port string, err error) {
+	host, port, err = net.SplitHostPort(hostPort)
+	if err != nil {
+		// Ignore "missing port in address" error as we use default port.
+		// This happens when hostPort = "" or hostPort = "server".
+		if !strings.Contains(err.Error(), "missing port in address") {
+			return "", "", err
+		}
+
+		host = hostPort
+		port = defaultPort
```

Comment:
no, because this will not work:

minio server --address :8000 http://host1/path1 http://host2/path2....

i.e we should consider 8000 and not 9000